### PR TITLE
Empty graph and quiet arg in transform_coords

### PR DESCRIPTION
**Pull request by jl-wynen**
The empty graph is useful for promoting an attr to coord. This can be used to partially revert an earlier transformation.

The `quiet` argument is useful when `transform_coords` is used as an implementation detail and the concrete transformation does not carry any meaning to the user.

### CODE REVIEW

File: docs/about/release-notes.rst [hunk 1]
```diff
@@ -15,6 +15,7 @@ Features
 * Switched to new system for runtime configuration, see the corresponding section in the reference documentation `#2389 <https://github.com/scipp/scipp/pull/2389>`_.
 * Added :py:func:`scipp.midpoints` which can be used to compute bin-centers `#2404 <https://github.com/scipp/scipp/pull/2404>`_.
 * Made it possible to use ``-1`` for one of the dimension sizes when using ``sc.fold`` `#2414 <https://github.com/scipp/scipp/pull/2414>`_.
+* Added ``quiet`` argument to :pu:func:`scipp.transform_coords` `#2420 <https://github.com/scipp/scipp/pull/2420>`_.
```
Comment (SimonHeybrock) [inline docs/about/release-notes.rst:18]:
```suggestion
* Added ``quiet`` argument to :py:func:`scipp.transform_coords` `#2420 <https://github.com/scipp/scipp/pull/2420>`_.
```

File: src/scipp/coords/transform_coords.py [hunk 2]
```diff
@@ -47,6 +48,8 @@ def transform_coords(x: Union[DataArray, Dataset],
                               Default is True.
     :param keep_inputs: Keep consumed input coordinates or attributes.
                         Default is True.
+    :param quiet: If True, no log output is produced. Otherwise, ``transform_coords``
+                  produces a log of its actions.
```
Comment (SimonHeybrock) [inline src/scipp/coords/transform_coords.py:52]: Mention the default?

Comment (jl-wynen) [reply src/scipp/coords/transform_coords.py:52]: Should be shown in the signature int he docs. See also this discussion: https://github.com/scipp/ess/pull/93#discussion_r794379578

There are certainly cases were it makes sense to repeat the default value. But I wouldn't do it too often. Is this case a case where we should mention it?

Comment (SimonHeybrock) [reply src/scipp/coords/transform_coords.py:52]: Sphinx sure, but what about `help(myfunc)`?

Comment (jl-wynen) [reply src/scipp/coords/transform_coords.py:52]: Shown but not great: (example for a version w/o `quiet`)
```python
>>> help(sc.transform_coords)
Help on function transform_coords in module scipp.coords.transform_coords:

transform_coords(x: Union[scipp._scipp.core.DataArray, scipp._scipp.core.Dataset], targets: Union[str, Iterable[str]], graph: Dict[Union[str, Tuple[str, ...]], Union[str, Callable]], *, rename_dims: bool = True, keep_aliases: bool = True, keep_intermediate: bool = True, keep_inputs: bool = True) -> Union[scipp._scipp.core.DataArray, scipp._scipp.core.Dataset]
    Compute new coords based on transformations of input coords.
    
...
```

The question mark in Jupyter is much better:
```python
[1]: sc.transform_coords?
Signature:
sc.transform_coords(
    x: Union[scipp._scipp.core.DataArray, scipp._scipp.core.Dataset],
    targets: Union[str, Iterable[str]],
    graph: Dict[Union[str, Tuple[str, ...]], Union[str, Callable]],
    *,
    rename_dims: bool = True,
    keep_aliases: bool = True,
    keep_intermediate: bool = True,
    keep_inputs: bool = True,
) -> Union[scipp._scipp.core.DataArray, scipp._scipp.core.Dataset]
Docstring:
Compute new coords based on transformations of input coords.

...
```

The problem is also that if we write the default value in the docstring, we have to keep it up to date. So I would be against doing it in all cases.

Comment (SimonHeybrock) [reply src/scipp/coords/transform_coords.py:52]: Ok.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**